src/Helper/HeadMeta.php

```diff
@@ -334,22 +335,27 @@ protected function isValid($item)
             return false;
         }
 
+        $doctype = $this->view->plugin('doctype');
+        if ($item->type === 'charset' && $doctype->isXhtml()) {
+            throw new Exception\RuntimeException('XHTML* doctype has no attribute charset; please use appendHttpEquiv()');
```

Comment:
`return false` here.

Reply:
already throw an Exception, need a return false ?

Reply:
understood, make a RuntimeException in `setCharset()` ?